core/payment/src/api/allocations.rs

```diff
@@ -162,12 +163,83 @@ async fn get_allocation(
     }
 }
 
+fn amend_allocation_fields(
+    old_allocation: Allocation,
+    new_allocation: NewAllocation,
+) -> Allocation {
+    Allocation {
+        allocation_id: old_allocation.allocation_id.clone(),
+        address: new_allocation
+            .address
+            .unwrap_or_else(|| old_allocation.address.clone()),
+        payment_platform: new_allocation
+            .payment_platform
+            .unwrap_or_else(|| old_allocation.payment_platform.clone()),
+        total_amount: new_allocation.total_amount.clone(),
+        spent_amount: old_allocation.spent_amount.clone(),
+        remaining_amount: new_allocation.total_amount - &old_allocation.spent_amount,
+        timestamp: old_allocation.timestamp,
+        timeout: new_allocation.timeout.or(old_allocation.timeout),
```

Comment:
Timeout should be validated if it isn't in the past

Reply:
Done